controllers/release/release_adapter.go

```diff
@@ -213,54 +214,59 @@ func (a *Adapter) EnsureSnapshotEnvironmentBindingIsCreated() (results.Operation
 		return results.ContinueProcessing()
 	}
 
-	err = a.syncResources()
+	environment, err := a.getEnvironment(releasePlanAdmission)
 	if err != nil {
 		return results.RequeueWithError(err)
 	}
 
-	binding, err := a.createOrUpdateSnapshotEnvironmentBinding(releasePlanAdmission)
+	err = a.syncResources()
 	if err != nil {
 		return results.RequeueWithError(err)
 	}
 
-	a.logger.Info("Created/updated SnapshotEnvironmentBinding",
-		"SnapshotEnvironmentBinding.Name", binding.Name, "SnapshotEnvironmentBinding.Namespace", binding.Namespace)
+	// Search for an existing binding
+	binding, err := a.getSnapshotEnvironmentBinding(environment, releasePlanAdmission)
+	if err != nil && !errors.IsNotFound(err) {
+		return results.RequeueWithError(err)
+	}
 
 	patch := client.MergeFrom(a.release.DeepCopy())
-	a.release.Status.SnapshotEnvironmentBinding = fmt.Sprintf("%s%c%s", binding.Namespace, types.Separator, binding.Name)
+
+	if binding == nil {
+		binding, err := a.createSnapshotEnvironmentBinding(releasePlanAdmission, environment)
+		if err != nil {
+			return results.RequeueWithError(err)
+		}
+
+		a.logger.Info("Created/updated SnapshotEnvironmentBinding",
+			"SnapshotEnvironmentBinding.Name", binding.Name, "SnapshotEnvironmentBinding.Namespace", binding.Namespace)
+
+		a.release.Status.SnapshotEnvironmentBinding = fmt.Sprintf("%s%c%s", binding.Namespace, types.Separator, binding.Name)
+	}
 
 	return results.RequeueOnErrorOrContinue(a.client.Status().Patch(a.context, a.release, patch))
 }
 
-// createOrUpdateSnapshotEnvironmentBinding creates or updates a SnapshotEnvironmentBinding for the Release being
-// processed.
-func (a *Adapter) createOrUpdateSnapshotEnvironmentBinding(releasePlanAdmission *v1alpha1.ReleasePlanAdmission) (*applicationapiv1alpha1.SnapshotEnvironmentBinding, error) {
-	application, components, snapshot, environment, err := a.getSnapshotEnvironmentResources(releasePlanAdmission)
-	if err != nil {
-		return nil, err
+// EnsureSnapshotEnvironmentBindingIsTracked is an operation that will ensure that the SnapshotEnvironmentBinding
+// Deployment status is tracked in the Release being processed.
+func (a *Adapter) EnsureSnapshotEnvironmentBindingIsTracked() (results.OperationResult, error) {
+	if !a.release.HasSucceeded() {
+		return results.ContinueProcessing()
 	}
 
-	// The binding information needs to be updated no matter if it already exists or not
-	binding := gitops.NewSnapshotEnvironmentBinding(components, snapshot, environment)
-
 	// Search for an existing binding
-	existingBinding, err := a.getSnapshotEnvironmentBinding(environment, releasePlanAdmission)
-	if err != nil && !errors.IsNotFound(err) {
-		return nil, err
+	binding, err := a.getSnapshotEnvironmentBindingFromReleaseStatus()
+	if err != nil {
+		return results.RequeueWithError(err)
```

Comment:
This forces the operation to work only with existing bindings which is not always the case. You should check if the error is a NotFound and work accordingly.

Alternatively and probably better, the first condition should look something like:
```
if release hasn't succeeded or status.binding is empty -> continue
```

To support this you could add a `IsBeingDeployed` function on the type. In addition, there's now a function, `HasBeenDeployed`, that has a terrible name taking in account what it does. This change sets `Status.SnapshotEnvironmentBinding` on binding creation time, so the logic should be changed to check for the condition status. This function should be used as a guard as well. If it was deployed already, no need to keep tracking the status.